src/protocolgame.h

```diff
@@ -317,6 +318,13 @@ class ProtocolGame final : public Protocol
 			g_dispatcher.addTask(createTask(delay, std::bind(std::forward<Callable>(function), &g_game, std::forward<Args>(args)...)));
 		}
 
+		struct Viewport {
+			int x = Map::maxClientViewportX;
+			int y = Map::maxClientViewportY;
+			int width = (x * 2) + 2;
+			int height = (y * 2) + 2;
+		} viewport;
+
```

Comment:
This is not configurable, just use `constexpr` values instead.

Reply:
What do you mean "no point in exporting it here."?

Reply:
Viewport struct in protocolgame class is redundant most likely. You can define it as static constexpr like most of the values at the top of the file.

Reply:
Now I understand, it really doesn't make sense.

Is that correct? 
```
int32_t VIEWPORT_X = Map::maxClientViewportX;
int32_t VIEWPORT_Y = Map::maxClientViewportY;
constexpr int32_t VIEWPORT_WIDTH = (VIEWPORT_X * 2) + 2;
constexpr int32_t VIEWPORT_HEIGHT = (VIEWPORT_Y * 2) + 2;
```

Reply:
I dont think we even need to redefine it, we have it already in map file, so I guess you can just use directly?

Reply:
Yes, using `Map::maxClientViewportX`

Ex:
```
GetMapDescription(oldPos.x - Map::maxClientViewportX, oldPos.y + (Map::maxClientViewportY + 1), newPos.z, 2 + (Map::maxClientViewportX * 2), 1, msg);
```
```
GetFloorDescription(msg, oldPos.x - Map::maxClientViewportX, oldPos.y - Map::maxClientViewportY, 4, 2 + (Map::maxClientViewportX * 2), 2 + (Map::maxClientViewportY * 2), 4, skip);
```

Reply:
> What do you mean "no point in exporting it here."?

Code inside a `.h` file is exported, all other files that import it will see. Since you only ever use this code inside `protocolgame.cpp`, it belongs in there, not here in the `.h`.

The `constexpr` part is what @nekiro explained